apps/studio/components/interfaces/DiskManagement/DiskManagementPanelForm.tsx

```diff
@@ -0,0 +1,600 @@
+  })
+  const mainDiskUsed = Math.round(((dbSize?.result[0].db_size ?? 0) / GB) * 100) / 100
+  const { includedDiskGB } = PLAN_DETAILS?.[planId as keyof typeof PLAN_DETAILS] ?? {}
+
+  const { mutate: updateDiskConfigurationRQ, isLoading: isUpdatingDiskConfiguration } =
+    useUpdateDiskAttributesMutation({
+      onSuccess: (_, vars) => {
+        toast.success(
+          'Successfully requested disk configuration changes! Your changes will be applied shortly'
+        )
+        const { ref, ...formData } = vars
+        setIsDialogOpen(false)
+        setRefetchInterval(3000)
+        form.reset(formData as DiskStorageSchemaType)
+      },
+    })
+
+  const defaultValues = {
+    storageType: type,
+    provisionedIOPS: iops,
+    throughput: throughput_mbps,
+    totalSize: size_gb,
+  }
+  const form = useForm<DiskStorageSchemaType>({
+    resolver: zodResolver(DiskStorageSchema),
+    defaultValues,
+    mode: 'onBlur',
+    reValidateMode: 'onChange',
+  })
+
+  const { watch, setValue, control, formState } = form
+  const watchedStorageType = watch('storageType')
+  const watchedTotalSize = watch('totalSize')
+  const watchedIOPS = watch('provisionedIOPS')
+  const { dirtyFields } = formState // Destructure dirtyFields from formState
+  const isAllocatedStorageDirty = !!dirtyFields.totalSize // Check if 'allocatedStorage' is dirty
+  const disableInput = isRequestingChanges || isPlanUpgradeRequired || isWithinCooldownWindow
+
+  const maxIOPS =
+    watchedStorageType === 'gp3'
+      ? Math.min(500 * watchedTotalSize, 16000)
+      : Math.min(1000 * watchedTotalSize, 256000)
+  const maxThroughput =
+    watchedStorageType === 'gp3' ? Math.min(0.25 * watchedIOPS, 1000) : undefined
+
+  const onSubmit = async (data: DiskStorageSchemaType) => {
+    if (projectRef === undefined) return console.error('Project ref is required')
+    updateDiskConfigurationRQ({ ref: projectRef, ...data })
+  }
+
+  const diskSizePrice = calculateDiskSizePrice({
+    planId,
+    oldSize: form.formState.defaultValues?.totalSize || 0,
+    oldStorageType: form.formState.defaultValues?.storageType as DiskType,
+    newSize: form.getValues('totalSize'),
+    newStorageType: form.getValues('storageType') as DiskType,
+  })
+
+  const iopsPrice = calculateIOPSPrice({
+    oldStorageType: form.formState.defaultValues?.storageType as DiskType,
+    oldProvisionedIOPS: form.formState.defaultValues?.provisionedIOPS || 0,
+    newStorageType: form.getValues('storageType') as DiskType,
+    newProvisionedIOPS: form.getValues('provisionedIOPS'),
+  })
+
+  const throughputPrice = calculateThroughputPrice({
+    storageType: form.getValues('storageType') as DiskType,
+    newThroughput: form.getValues('throughput') || 0,
+    oldThroughput: form.formState.defaultValues?.throughput || 0,
+  })
+
+  useEffect(() => {
+    // Initialize field values properly when data has been loaded
+    if (isSuccess) form.reset(defaultValues)
+  }, [isSuccess])
```

Comment:
oh yeah you're right, this useEffect is unnecessary - will remove

Reply:
hmm... actually thinking might need this. if i put this in onSuccess its causing the following issue:

Divit ran into an issue that i couldn't reproduce such that the form value resets when he switches tabs?
i'm thinking its because RQ is just revalidating and triggers onSuccess, whereas I'd only want to set the form once (which can be achieved with the useEffect since the value of isSuccess doesnt change)

Reply:
actually we might need this useEffect, was previously thinking the onSuccess was sufficient

Divit is running into an issue that i could repro, such that if we edit the form, go to another tab and wait a bit then come back to the same tab, I think RQ revalidates and triggers the onSuccess, hence resetting the form (we only want the form to reset once at page load, and if we were waiting for the requested changes to be enacted)

if we use the useEffect then we can guarantee it runs only once, cause the value of isSuccess will not change